### ISSUE TITLE
Draft sockets do not work when building from vcpkg

### ISSUE THREAD

**Issue by AHernandez-BGS** [zeromq/czmq#2034]
Hello,

I noticed that Radio/Dish sockets would return null when attempting to create them. When I investigated further I noticed that in zsock.c the draft sockets do the following:

```
#if defined ZMQ_DISH
    zsock_t *sock = zsock_new_checked (ZMQ_DISH, filename, line_nbr);
    if (zsock_attach (sock, endpoints, false))
        zsock_destroy (&sock);
    return sock;
#else
    errno = ENOTSUP;
    return NULL;
#endif
```

The issue (from what I can tell) is that running the following in vcpkg:
`vcpkg.exe install czmq[draft]`
will correctly build zeromq with drafts enabled- however, the czmq project compiles without the **ZMQ_BUILD_DRAFT_API** definition. The problem is that **ZMQ_DISH** (et al draft sockets) are not available when that definition is missing. So JUST the internal draft socket setup does not make it into the binaries.

I've worked around this locally by modifying czmq's cmakelist in my vcpkg buildtree:
line 54:
```
IF (ENABLE_DRAFTS)
    ADD_DEFINITIONS (-DZMQ_BUILD_DRAFT_API)    #<ADDED THIS
    ADD_DEFINITIONS (-DCZMQ_BUILD_DRAFT_API)
ENDIF (ENABLE_DRAFTS)
```

everything is working now but according to #1475 it was mentioned that we should NOT work around things in czmq. I was curious what the correct fix would be for anyone else using vcpkg?

**Comment by AHernandez-BGS** [comment]
For reference, here's a snippet of the defines from zmq.h:

```
#ifdef ZMQ_BUILD_DRAFT_API

/*  DRAFT Socket types. */
#define ZMQ_SERVER 12
#define ZMQ_CLIENT 13
#define ZMQ_RADIO 14
#define ZMQ_DISH 15
#define ZMQ_GATHER 16
#define ZMQ_SCATTER 17
#define ZMQ_DGRAM 18
<snip>
#endif // ZMQ_BUILD_DRAFT_API
```

**Comment by bluca** [comment]
if vcpkg uses the pkgconfig file created by libzmq, it should get the variable defined automatically

**Comment by AHernandez-BGS** [comment]
I'll verify what's happening there tonight and followup 👍

**Comment by AHernandez-BGS** [comment]
I've traced the vcpkg build steps of zmq and czmq. All I can tell is that CMake is configuring the respective libzmq.pc.in and libczmq.pc.in configuration files. However, all of the captured build logs indicate that CMake is doing all of the work, pkgconfig is new to me so i'm not sure how to bridge the gap there...

Hopefully someone smarter than me can spot any potential issues :)

In the meantime I've elected to implement the necessary app behaviors using non-draft sockets. Hopefully I can afford more time in the future to revisit the issue.
